### PR TITLE
Fix releasing

### PR DESCRIPTION
**Pull request by hasufell**
Tested here: https://github.com/hasufell/haskell-language-server/releases

----

The idea here is that we pull the archives from the gitlab API instead of implementing pushing. That means a pipeline with the same tag as the release will have to have succeeded fully on gitlab CI.

----

Related: https://github.com/haskell/haskell-language-server/issues/2663

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/2707"></a>

### CODE REVIEW

File: .github/workflows/build.yml
Comment (hasufell) [inline .github/workflows/build.yml:18]: This well need to be changed to `1180`, which is the project id of https://gitlab.haskell.org/haskell/haskell-language-server

I set this to my fork for testing.

File: .github/workflows/build.yml
Comment (hasufell) [inline .github/workflows/build.yml:81]: This is also just set for testing... should be `ver=${tag/v/}`  as below

File: .github/workflows/build.yml
Comment (hasufell) [inline .github/workflows/build.yml:76]: testing value

File: .github/workflows/build.yml
Comment (hasufell) [inline .github/workflows/build.yml:124]: Because `upload-release-asset` does not support wildcards

**Comment by hasufell** [comment]
@jneira

**Comment by jneira** [comment]
sorry dont have time to review the pr, surely some other maintainer(s) could do it

**Comment by Anton-Latukha** [comment]
Currently reviewed it 2 times (had 2 passes). The second one - read the code.

I am not adept in AWK language.

Honestly - to understand PR better structurally (where what how goes) & run inference inside - the third & maybe fourth passes are needed.

Of course that it is a single commit with a big Bourne shell diff. I understand - subject matter makes making separate commits difficult, but it also becomes harder to review & internalize what was done precisely & the meaning of it. It is a long way of saying - to give a proper review - I need another pass. Seems pretty straightforward & understandable, though.

Seen that you've done testing during the work process.

I can review better & say/not say something more meaningful later in the day.

**Comment by hasufell** [comment]
> I am not adept in AWK language.

That's just some foo that I might remove later, because the job names on gitlab differ from the tarball names (and we need both for download via API). That can be equalized, but requires another round of gitlab CI, which I wanted to skip for testing.

> Seems pretty straightforward & understandable, though.

Yeah, we just drop the static builds from github CI, pull all artifacts from gitlab and then upload them to the github release. We use this rather convoluted github `script` action, because the [upload-release-asset action](https://github.com/actions/upload-release-asset) is deprecated and doesn't support wildcards.

**Comment by Anton-Latukha** [comment]
No. It is mine issue. I just as many was postponing learning AWK for years. At this time & level I probably need to know some AWK.

Yes, I understood that it is just reshuffling of fields, between probably job & the tarball names.

**Comment by Anton-Latukha** [comment]
> We use this rather convoluted github script action, because the [upload-release-asset action](https://github.com/actions/upload-release-asset) is deprecated and doesn't support wildcards.

Thank you for complete clarification.

As [upload-release-asset action](https://github.com/actions/upload-release-asset) mentions was https://github.com/softprops/action-gh-release considered instead? It may be looked into after the PR merge.

**Comment by Anton-Latukha** [comment]
The link to the resulting workflow run is: https://github.com/hasufell/haskell-language-server/runs/5169934711?check_suite_focus=true

File: .github/workflows/build.yml
Comment (Anton-Latukha) [inline .github/workflows/build.yml:71]: Oh.

Thank you for setting the `x` here, great idea.

File: .github/workflows/build.yml
Comment (Anton-Latukha) [inline .github/workflows/build.yml:140]: Probably some mostly copy-paste JS script to replace the cause.

**Review by Anton-Latukha — APPROVED** [review]
Reviewed.

The situation & solution is understandable and straightforward.

Only had a couple of the most minimal improvements to the code, but the code is overall of high quality.

The use of the `set -x` in the shell script would help when some naming changes or API changes. I think for the prolonged shell parts it is a good idea to keep the `x` flag on.

With that https://github.com/hasufell/haskell-language-server/runs/5169934711?check_suite_focus=true works & artifacts are uploaded - I think my review is enough to mandate the merge.

**Comment by Anton-Latukha** [comment]
[skip circleci]

**Comment by Anton-Latukha** [comment]
@hasufell, should we report to evaluate https://github.com/softprops/action-gh-release for later?

**Comment by hasufell** [comment]
> @hasufell, should we report to evaluate use of https://github.com/softprops/action-gh-release later, or it was considered/does not implement the requirements.

Afaiu that action creates a release, which we don't want. We create a release and trigger a build instead. I don't think it's worth investigating that action.

**Comment by Anton-Latukha** [comment]
Thank you @hasufell for this.

**Pull request merged**
